src/lib/divider/divider.spec.ts

```diff
@@ -0,0 +1,99 @@
+import {async, TestBed} from '@angular/core/testing';
+import {Component} from '@angular/core';
+import {By} from '@angular/platform-browser';
+import {MatDividerModule} from './divider-module';
+
+
+describe('MatDivider', () => {
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [MatDividerModule],
+      declarations: [
+        MatDividerBase,
+        MatDividerVertical,
+        MatDividerInset,
+        MatDividerVerticalInset,
+        MatDividerWithCssClass,
+      ],
+    });
+
+    TestBed.compileComponents();
+  }));
+
+  it('should not apply any additional class to a list without lines', () => {
+    const fixture = TestBed.createComponent(MatDividerBase);
+    const divider = fixture.debugElement.query(By.css('mat-divider'));
+    fixture.detectChanges();
+    expect(divider.nativeElement.className).toBe('mat-divider');
+  });
+
+  it('should apply vertical class to vertical divider', () => {
+    const fixture = TestBed.createComponent(MatDividerVertical);
+    fixture.detectChanges();
+
+    const divider = fixture.debugElement.query(By.css('mat-divider'));
+    expect(divider.nativeElement.className).toContain('mat-divider');
+    expect(divider.nativeElement.className).toContain('mat-divider-vertical');
+  });
+
+  it('should apply inset class to inset divider', () => {
+    const fixture = TestBed.createComponent(MatDividerInset);
+    fixture.detectChanges();
+
+    const divider = fixture.debugElement.query(By.css('mat-divider'));
+    expect(divider.nativeElement.className).toContain('mat-divider');
+    expect(divider.nativeElement.className).toContain('mat-divider-inset');
+  });
+
+  it('should apply inset and vertical classes to vertical inset divider', () => {
+    const fixture = TestBed.createComponent(MatDividerVerticalInset);
+    fixture.detectChanges();
+
+    const divider = fixture.debugElement.query(By.css('mat-divider'));
+    expect(divider.nativeElement.className).toContain('mat-divider');
+    expect(divider.nativeElement.className).toContain('mat-divider-inset');
+    expect(divider.nativeElement.className).toContain('mat-divider-vertical');
+  });
+
+  it('should not clear custom classes provided by user', () => {
+    const fixture = TestBed.createComponent(MatDividerWithCssClass);
+    fixture.detectChanges();
+
+    const divider = fixture.debugElement.query(By.css('mat-divider'));
+    expect(divider.nativeElement.classList.contains('test-class')).toBe(true);
+  });
+
+  it('should add aria roles properly', () => {
+    const fixture = TestBed.createComponent(MatDividerBase);
+    fixture.detectChanges();
+
+    const divider = fixture.debugElement.query(By.css('mat-divider'));
+    expect(divider.nativeElement.getAttribute('role')).toBe('separator');
+  });
+});
+
+@Component({
```

Comment:
Rather than creating a number of Components that have to be recompiled on every run, can you create a single Component that is altered through attributes for tests?

We have found a bottleneck in our tests comes from having to recompile many components on each test run.

```ts
@Component({
  template: `<mat-divider [inset]="inset" [vertical]="vertical"></mat-divider>`
})
class MatDividerTestComponent {
  vertical: boolean;
  inset: boolean;
}
```

Then in tests you would do:
```ts
it('should apply a vertical class', () => {
  fixture.componentInstance.vertical = true;
  fixture.detechChanges();

  expect(divider.nativeElement.className).toContain('mat-divider-vertical');
});

Reply:
Great suggestions, will add